optuna/distributions.py

```diff
@@ -89,10 +91,91 @@ def __hash__(self) -> int:
 
     def __repr__(self) -> str:
 
-        kwargs = ", ".join("{}={}".format(k, v) for k, v in sorted(self.__dict__.items()))
+        kwargs = ", ".join("{}={}".format(k, v) for k, v in sorted(self._asdict().items()))
         return "{}({})".format(self.__class__.__name__, kwargs)
 
 
+class FloatDistribution(BaseDistribution):
+    """A continuous uniform distribution in a linear domain.
```

Comment:
By specifying `step`, the distribution becomes discrete. In addition, when we use `log=True`, the domain is log-scale, not linear domain. How about this?

```suggestion
    """A distribution on floats.
```

I'm not sure this is the optimal explanation. 
What do you think about it?

Reply:
Thank you for pointing out! I agree that the class is much more versatile than my explanation. Let me update as you suggested.